### ISSUE TITLE
Indicator doesn't show when used with --grace

### ISSUE THREAD

**Issue by jmederosalvarado** [mortie/swaylock-effects#60]
I'm using the swaylock-effecta-git from AUR on Arch Linux. When I set a grace period, indicator never shows up, however if I run the exact same command without the grace option everything works as expected.

**Comment by mortie** [comment]
What's the exact command you're running?

With `--indicator`, the indicator is supposed to always be visible. However, with `--indicator-idle-visible`, the indicator is supposed to show up only after the grace period. Could what you're seeing just be a result of using `--indicator-idle-visible`?

What's your `swaylock --version`?

**Comment by jmederosalvarado** [comment]
Hello. Thanks for the quick response. The exact command I'm using is:

```bash
swaylock \
    --daemonize \
    --screenshots \
    --clock \
    --indicator-idle-visible \
    --indicator-radius 100 \
    --indicator-thickness 5 \
    --effect-scale 0.5 \
    --effect-blur 7x5 \
    --effect-scale 2 \
    --effect-vignette 0.5:0.5 \
    --ring-color bb00cc \
    --key-hl-color 880033 \
    --line-color 00000000 \
    --inside-color 00000088 \
    --separator-color 00000000 \
    --fade-in 0.5 \
    --grace 2
```

**swaylock version**: v1.6-3-11-g5cb9579 (Jun 22 2021, branch 'master')

As I understand it this command should show the indicator after the 2 seconds grace period. Am I correct?. However this does not happen, the indicator is never shown, when used without the `--grace` option, the indicator is shown from the beginning as I would expect.

Please let me know if there's anything else I could help with.

**Comment by mortie** [comment]
Hmm, everything looks correct. On the exact same swaylock-effects version, with the same command, I see the indicator immediately after the 2 second grace period.

Since I can't reproduce it, this is kind of a shot in the dark, but could you show me the output of your command if you add `--debug --trace` to it? Also, what version of Sway are you using?

**Comment by jmederosalvarado** [comment]
This is the output from the same command using `--debug` and `--trace`. Everything looks normal to me.

```
2021-06-22 10:00:04 - [main.c:1780] Parsing CLI Args
2021-06-22 10:00:04 - [main.c:427]: trace: handle_wl_output_geometry
2021-06-22 10:00:04 - [main.c:447]: trace: handle_wl_output_scale
2021-06-22 10:00:04 - [main.c:195]: trace: daemonize_start
2021-06-22 10:00:04 - [main.c:651]: trace: handle_xdg_output_name
2021-06-22 10:00:04 - [main.c:652] output name is eDP-1
2021-06-22 10:00:04 - [main.c:664]: trace: handle_xdg_output_done
2021-06-22 10:00:04 - [main.c:350]: trace: layer_surface_configure
2021-06-22 10:00:04 - [main.c:520]: trace: handle_screencopy_frame_buffer
2021-06-22 10:00:04 - [main.c:547]: trace: handle_screencopy_frame_flags
2021-06-22 10:00:04 - [main.c:595]: trace: handle_screencopy_frame_ready
2021-06-22 10:00:04 - [main.c:614] Loaded screenshot for output eDP-1
2021-06-22 10:00:04 - [main.c:331] Surface for output eDP-1 ready
2021-06-22 10:00:05 - [main.c:227]: trace: daemonize_done
2021-06-22 10:00:20 - [main.c:227]: trace: daemonize_done
```

**sway version**: 1.6

**Comment by jmederosalvarado** [comment]
Wow. I took a long shot here and I think I narrowed it down. It turns out that when I connect my second monitor, which I had not used in a while, it works as expected, it shows the indicator in the second monitor.

I don't know if this is actually expected though.

**Comment by jmederosalvarado** [comment]
Hi, so after reading through #53 I tried to set my display scaling to 1.0, and this indeed fixes the issue. So it appears that the problem only occurs when the display scaling is different from 1.0, this is consistent with the results I got when I plugged in my external monitor, as the scaling of this one is 1.0. As stated in the referenced issue, the problem only occurs when the indicator is not forced to be shown by using some flag like `--indicator` or when used with `--grace`.

I really don't know if we should close this and mark it as a duplicate of #53, I believe the underlying problem is the same, but I don't think it's the same issue. Please let me know how I could help.
